### DISCUSSION TITLE
map null input collection to an empty collection

### DISCUSSION THREAD

**Discussion by deweyjose** [mapstruct/mapstruct#2698] [General]
Hi team, thanks for providing such a useful library.

I'm curious if there's a way to map a null collection input value to an equivalent empty collection return type?
Not at the @Mapping/field level but at the public interface level. 

```
@Mapper
public interface foo {
  List<Bar> mapFromListFoo(List<Foo> in);
}


List<Foo> x = null;

var bars = foo.mapFromListFoo(x)

// bars == essentially List.of() or Collections.emptyList()
```

**Comment by filiphr** [comment]
This can be controlled through the `NullValueMappingStrategy`. Starting from 1.5.0.Beta2 you can control beans / collections separately. Have a look at the [NullValueMappingStrategy for maps / collections](https://mapstruct.org/news/2021-12-12-mapstruct-1_5_0_Beta2-is-out/#nullvaluemappingstrategy-for-maps--collections) for the separate control and [Mapping result for null arguments](https://mapstruct.org/documentation/stable/reference/html/#mapping-result-for-null-arguments) for what is in the latest Final release.
